src/cromshell/utilities/cromshellconfig.py

```diff
@@ -0,0 +1,130 @@
+import logging
+import json
+from pathlib import Path
+import csv
+
+LOGGER = logging.getLogger(__name__)
+
+"""Setup Cromshell config details. Intended to be used as a singleton"""
+
+# Set Cromshell Configuration Default Values
+metadata_parameters = "excludeKey=submittedFiles&expandSubWorkflows=true"
+slim_metadata_parameters = (
+    "=includeKey=id&includeKey=executionStatus&includeKey=backendStatus&includeKey"
+    "=status&includeKey=callRoot&expandSubWorkflows=true&includeKey"
+    "=subWorkflowMetadata&includeKey=subWorkflowId"
+)
+api_string = "/api/workflows/v1/"
+
+
+def override_slim_metadata_parameters(slim_metadata_param):
+    """Override Cromwell Slim Metadata Key From Command Line"""
+
+    global slim_metadata_parameters
+
+    if slim_metadata_param:
+        slim_metadata_parameters = (
+                slim_metadata_param
+                + "&includeKey=subWorkflowMetadata&includeKey=subWorkflowId"
+        )
+
+
+def resolve_cromwell_config_server_address(server_user=None, workflow_id=None):
+    """
+    Override Cromwell Server From Command Line or Environment or Submission file
+
+    This function allows users to override the cromwell server set in the
+    cromshell config file using the command line options or environment
+    variables. By default the server URL in the config file is used, however
+    if an URL is specified in the command line or environment then this
+    overrides the default. If a workflow Id is provided without a server URL
+    then the server associated with a workflow id in the submission file
+    overrides the default. If a workflow ID is provided but isn't found
+    in the submission file then the default server is used.
+    """
+
+    global cromwell_server
+    global submission_file
```

Comment:
Ah.   I initially misunderstood these variables.  Can you add definitions for these at the top of this file to make it clear they are module-level variables for `cromshellconfig`?

Reply:
Wait, I may be missing something.  Where are these initially defined?

Reply:
I found them!  Can you add a comment above these that says where they are defined?  I take for granted that they'll be at the top of the file so I got confused by their reference here.

Reply:
Defined variables as `None` at the top with a comment saying they will be set after functions have been defined

Reply:
👍